tests/func/experiments/test_checkpoints.py

```diff
@@ -188,3 +194,103 @@ def test_resume_non_head_checkpoint(
     )
     new_head = first(results)
     assert orig_branch != dvc.experiments.get_branch_by_rev(new_head)
+
+
+@pytest.fixture
+def clear_env(monkeypatch):
+    yield
+    monkeypatch.delenv(DVC_EXP_GIT_REMOTE, raising=False)
+    monkeypatch.delenv(DVC_EXP_AUTO_PUSH, raising=False)
+
+
+@pytest.mark.parametrize("use_url", [True, False])
+def test_auto_push_during_iterations(
```

Comment:
This test can probably be simplified. For this test case, what's important is whether or not we automatically push when the env vars are set. We don't actually need to test what happens in `dvc.experiments.push` (since we already have other tests for that behavior).

I think all we really need to do here is `mocker.spy` `experiments.push` to confirm that it is called the expected number of times, with the expected arguments.

So if the env var is enabled, push should be called once per commit - including the final non-checkpoint commit) with the arguments we expect (the correct git URL, and with DVC cache + run cache set to true).

If the env var is disabled, it should not be called at all.

Reply:
Good suggestion, before this I used to plan to add a new test case which will exit in running.